RecoEgamma/PhotonIdentification/src/PhotonMVABasedHaloTagger.cc

```diff
@@ -0,0 +1,425 @@
+ 2. JetMET POG: https://indico.cern.ch/event/1027614/contributions/4314949/attachments/2224472/3767396/beamHalo_12April.pdf
+ */
+
+#include "RecoEgamma/PhotonIdentification/interface/PhotonMVABasedHaloTagger.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHit.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h"
+#include <DataFormats/Math/interface/deltaPhi.h>
+#include "CommonTools/MVAUtils/interface/GBRForestTools.h"
+
+PhotonMVABasedHaloTagger::PhotonMVABasedHaloTagger(const edm::ParameterSet& conf, edm::ConsumesCollector&& iC)
+    : geometryToken_(iC.esConsumes()),
+      ecalPFRechitThresholdsToken_(iC.esConsumes()),
+      ecalClusterToolsESGetTokens_(std::move(iC)) {
+  rhoLabel_ = iC.consumes<double>(conf.getParameter<edm::InputTag>("rhoLabel"));
+
+  EBecalCollection_ = iC.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("barrelEcalRecHitCollection"));
+  EEecalCollection_ = iC.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("endcapEcalRecHitCollection"));
+  ESCollection_ = iC.consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("esRecHitCollection"));
+  HBHERecHitsCollection_ = iC.consumes<HBHERecHitCollection>(conf.getParameter<edm::InputTag>("HBHERecHitsCollection"));
+  recHitEThresholdHB_ = conf.getParameter<EgammaHcalIsolation::arrayHB>("recHitEThresholdHB");
+  recHitEThresholdHE_ = conf.getParameter<EgammaHcalIsolation::arrayHE>("recHitEThresholdHE");
+
+  noiseThrES_ = conf.getParameter<double>("noiseThrES");
+}
+
+double PhotonMVABasedHaloTagger::calculateMVA(const reco::Photon* pho,
+                                              const GBRForest* gbr_,
+                                              const edm::Event& iEvent,
+                                              const edm::EventSetup& es) {
+  bool isEB = pho->isEB();
+
+  if (isEB)
+    return 1.0;  /// this MVA is useful and trained only for the EE photons. For EB, there are a lot of other useful handles which can reject beam halo efficiently
+
+  //rho handle
+  double rho_ = iEvent.get(rhoLabel_);
+
+  // Get all the RecHits
+  const auto& ecalRecHitsBarrel = iEvent.get(EBecalCollection_);
+  const auto& ecalRecHitsEndcap = iEvent.get(EEecalCollection_);
+  const auto& esRecHits = iEvent.get(ESCollection_);
+  const auto& hbheRecHits = iEvent.get(HBHERecHitsCollection_);
+
+  //gets geometry
+  pG_ = es.getHandle(geometryToken_);
+  const CaloGeometry* geo = pG_.product();
+
+  ///ECAL PF rechit thresholds
+  auto const& thresholds = es.getData(ecalPFRechitThresholdsToken_);
+
+  noZS::EcalClusterLazyTools lazyToolnoZS(
+      iEvent, ecalClusterToolsESGetTokens_.get(es), EBecalCollection_, EEecalCollection_);
+
+  ///calculate the energy weighted X, Y and Z position of the photon cluster
+  if (isEB)
```

Comment:
Does this reman in view of a possible future usage of this tool also for photons in the barrel? [As of now](https://github.com/cms-sw/cmssw/pull/36901/files#diff-bcdf596b8b859dd2a3e92d3b48535675174c00912d5c2f4f43e574a307a650baR40-R41), this is dead code

Reply:
yes this is just in view of any future developments using a similar approach HB/HE and EE